prefetch.bs

```diff
@@ -252,297 +324,315 @@ A <dfn export>cross-origin prefetch IP anonymization policy</dfn> has an <dfn ex
-            </div>
-        1. Let |hypotheticalPartitionKey| be the result of [=determining the network partition key=] given |environment|.
-        1. If |originalPartitionKey| is equal to |hypotheticalPartitionKey|, then return "`Blocked`".
+                    <div class="note">This copy is complete before continuing, in the sense that subresource fetches, {{Document/cookie|document.cookie}}, etc. can observe the cookies.</div>
+            1. Otherwise:
+                1. Let |coopEnforcementResult| be the result of [=creating a cross-origin opener policy enforcement result for navigation=] given <var ignore>navigable</var>'s [=navigable/active document=] and <var ignore>entry</var>'s [=session history entry/document state=]'s [=document state/initiator origin=].
+                1. Set <var ignore>navigationParams</var> to the result of [=creating navigation params by fetching=] given |request|, <var ignore>entry</var>, |coopEnforcementResult|, <var ignore>navigable</var>, <var ignore>sourceSnapshotParams</var>, <var ignore>targetSnapshotParams</var>, <var ignore>cspNavigationType</var>, <var ignore>navigationId</var>, and <var ignore>navTimingType</var>.
 
-            <div class="note">The prefetch would end up in the same partition as the prefetch came from in this case. |environment| represents the environment that would ordinarily be used during navigation fetch.</div>
-        1. If there are [=credentials=] associated with |request|'s [=request/current URL=] and |hypotheticalPartitionKey|, then [=set/append=] |request|'s [=request/current URL=]'s [=url/origin=] to |originsWithConflictingCredentials|.
-        1. If |request|'s [=request/URL=] is not [=potentially trustworthy URL|potentially trustworthy=], then return "`Blocked`".
+</div>
 
-            <div class="note">This is intended to both reduce the likelihood of prefetch traffic being visible to an on-path attacker, and to encourage the use of cryptographic schemes over public networks.</div>
-        1. If |request| cannot be fetched given |anonymizationPolicy| for an [=implementation-defined=] reason, then return "`Blocked`".
+<hr>
 
-            <div class="note">This explicitly acknowledges that implementations might have additional restrictions. For instance, anonymized traffic might not be possible to some hosts, such as those that are not publicly routable and those that have <a href="https://buettner.github.io/private-prefetch-proxy/traffic-advice.html">traffic advice</a> declining private prefetch traffic.
-        1. [=redirect chain/Append=] |request| to |prefetchRecord|'s [=prefetch record/redirect chain=].
-        1. Return "`Allowed`".
+Add the following [=struct/item=] to [=session history entry=]:
+* <dfn for="session history entry">prefetch record</dfn>, a [=prefetch record=] or null. If non-null, it indicates that a prefetch, not a normal navigation, is in progress.
```

Comment:
It feels weird to store this in session history entry. It seems like it doesn't need to be there long term, and instead is just used because the relevant algorithms take a SHE. Can we instead just thread it to the relevant algorithms as an independent, sometimes-null parameter?

Reply:
The entire SHE here is admittedly a little fictitious because we're not actually putting it in session history just yet.

Made the suggested change.